General update of docs

axsaucedo opened the pull request:
Big update for docs, including:
* Moving examples into categories
* Added make command to build docs faster with livereload (skipping examples and python api)
* Updated readme to include example and latest content
* Removed unused documentation pages
* Added Jenkins X and Jenkins Classic examples from separate repo
* Added page on kubeflow
* Added page on CI / CD MLOps
* Added linked Changelog
* Added documentation to helm chart itself and linked it to docs
* Added RST files to link pages
* Removed Old MLFlow example

seldondev commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **axsaucedo**

If they are not already assigned, you can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["axsaucedo"]} -->

seldondev commented:
Sun Mar 15 21:39:05 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1540/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1540 --build=1

seldondev commented:
Sun Mar 15 21:40:26 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1540/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1540 --build=2

cliveseldon commented:
We should also remove examples/outliers/alibi-detect-combiner/notebook.ipynb until the new way of doing it is completed.

cliveseldon commented:
In Specializes framwork examples: should we place into "incubating" 
 * kubeflow e2e pipeline
 * H2O MoJo 

until they are updated?

cliveseldon commented:
We should update the blogs section, but can do that in a separate issue?

cliveseldon commented:
html/analytics/custom_metrics.html is not in correct section as its the valid cusotm metrics docs for python. Also - this is what I am adding in the PR with @RafalSkolasinski : a new example.

Maybe put back to where it was and remove example link. We can rebase when this is in and put into docs.

cliveseldon commented:
analytics/kubeflow.html is a cut and paste from kubeflow docs and doesn't read well.
Also not sure this is the best location in "production"

cliveseldon commented:
Why is Java language wrapper not moved to incubating section?

cliveseldon commented:
"Distributed Tracing" has gone missing from index

cliveseldon commented:
This is a bad title "Deployment Options running Seldon Core with Ambassador"

cliveseldon commented:
Why is 1.0.2 not in changelog?

cliveseldon commented:
Lots of examples are still v1alpha2 - can we fix to v1 in all docs.

cliveseldon commented:
Can you check the release script respects comments in operator helm value.yaml as it recreates it?

cliveseldon commented on doc/source/analytics/logging.md:
```diff
@@ -44,6 +44,26 @@ The specification is:
  * url: Any url. Optional. If not provided then it will default to the default knative borker in the namespace of the Seldon Deployment.
  * mode: Either `request`, `response` or `all`
 
+# Setting Global Default
```
Should be `##` as this messes up the index when its top level `#`

axsaucedo replied:
Ok changed

cliveseldon commented on doc/source/workflow/quickstart.md:
````diff
@@ -0,0 +1,309 @@
+[Created file at /mypath/model.pickle]
+```
+
+**2. Create a wrapper class `Model.py`**
````
Seem not to make bold

axsaucedo replied:
Removed characters so it makes bold

axsaucedo commented on doc/source/developer/community.md:
```diff
@@ -0,0 +1,32 @@
+
+**Dates & times**
+
+The dates for the community call is every Thursday at 16:00 UK Time.
```
Good catch

axsaucedo commented:
@cliveseldon addressed most, here are some comments - current outstanding is checking release script and checking if alibi example can be updated.

>  We should also remove examples/outliers/alibi-detect-combiner/notebook.ipynb until the new way of doing it is completed.

Ok cool. We could also move it to Incubating given that I guess it's still being explored

> I think we should remove examples/explainers/alibi_anchor_tabular/alibi_anchor_tabular_seldon_deployment.ipynb as this is not how we do explanations anymore?

I think it would be quite important to show how to wrap a custom explainer, I have updated it but it doesn't seem to work with models built with the s2i wrapper, is this something by design?


> We should update the blogs section, but can do that in a separate issue? 

Sounds good, I've opened #1550

>  analytics/kubeflow.html is a cut and paste from kubeflow docs and doesn't read well.

Yeah it was a literal cut and paste from their markdown - I've removed the examples at the bottom, what else could we add? I couldn't find the instructions to add the kfapp config, happy to add extra pieces

>  Also not sure this is the best location in "production"

I agree, maybe once we finalise the MLFlow integration, we coudl have a section where we could have another section with pages on our integration with MLFlow, Pachyderm, Kubeflow, etc?

>  Why is Java language wrapper not moved to incubating section?

I think it was because python seemed a bit awkward on the section by itself, but I've updated it now.

>  This is a bad title "Deployment Options running Seldon Core with Ambassador"

Changed to "Ingress options with ambassador"

>  Why is 1.0.2 not in changelog?

It's reference to CHANGELOG.md, so it's may just not be there

>  Lots of examples are still v1alpha2 - can we fix to v1 in all docs.

It may be better in another PR as it seems it may take some time to go through al examples

>  Can you check the release script respects comments in operator helm value.yaml as it recreates it?

Ok - I need to check this one

seldondev commented:
Mon Mar 16 15:55:20 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1540/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1540 --build=3

seldondev commented:
Mon Mar 16 15:56:27 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1540/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1540 --build=4

seldondev commented:
Mon Mar 16 22:04:09 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1540/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1540 --build=5

seldondev commented:
Mon Mar 16 22:04:15 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1540/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1540 --build=6

axsaucedo commented:
@cliveseldon I've resolved the lsat two pieces - I've removed the alibi example for now. I can also confirm that the Release script actually does not keep the comments... Perhaps it would be good to hear what @gsunner thinks? Basically I've added comments on the values.yaml file for self documentation, but the release script overrides all the comments, is that something that will be saved upon release, or do we keep the original values.yaml? Here is the file: https://github.com/SeldonIO/seldon-core/blob/da987a6da1559a5fe0e5718b42d8dac2c29b5e7d/helm-charts/seldon-core-operator/values.yaml

cliveseldon commented:
/approve

seldondev commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1540#issuecomment-600218380" title="Approved">cliveseldon</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
